ts/IterableInbox.tsx

```diff
@@ -109,6 +111,42 @@ const IterableInbox = ({
       }
    }, [width])
 
+   if(appState === 'active') {
+      if(isFocused) {
+         inboxDataModel.startSession()
+         inboxDataModel.updateVisibleRows(visibleRowImpressions)
+      } else {
+         inboxDataModel.endSession()
+      }
+   }
+
+   function getInitialMessageImpressions(messageListHeight: number, messageRowHeight: number, rowViewModels: InboxRowViewModel[]) {
+      let numMessages = Math.floor(messageListHeight / messageRowHeight)
```

Comment:
I think this way of calculating the on screen messages is going to be too fragile to use. Is there a way we can skip depending on calculations like this? An easy this would break would be if individual messages have different heights, like different kinds of messages (and subsequently, different heights than the standard) in an inbox.

Reply:
Yes, I did more research and found a way to trigger the onViewableItemsChanged using recordInteraction(). see [here](https://reactnative.dev/docs/flatlist).